pkg/identityd2/storage/storage_impl.go

```diff
@@ -109,7 +111,38 @@ func (self *StorageImpl) DeleteDomain(id string) error {
 }
 
 func (self *StorageImpl) PatchDomain(id string, domain *Domain) (*Domain, error) {
-	panic("unimplemented")
+	var err error
+	var dom Domain
+
+	tx := self.db.Begin()
+
+	if err = tx.First(&dom, "Id = ?", id).Error; err != nil {
+		return nil, err
+	}
+	if domain.Name != nil && dom.Name != domain.Name {
```

Comment:
```suggestion
	if domain.Name != nil {
```
我觉得既然已经用了指针来表达是否修改的话, 就不用再对比两个变量是否一样了吧?